### PR TITLE
[GPTQ] Fix test

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ? 
This PR fixes failing tests related to GPTQ quantization. The breaking tests are related to modification on optimum side and OOM from the new runner. I've also replaced for a smaller model. related optimum [PR](https://github.com/huggingface/optimum/pull/1574/files#)

### CODE REVIEW

File: tests/quantization/gptq/test_gptq.py
Comment (amyeroberts) [inline tests/quantization/gptq/test_gptq.py:250]: Why remove this line?

File: tests/quantization/gptq/test_gptq.py
Comment (amyeroberts) [inline tests/quantization/gptq/test_gptq.py:245]: Why remove this line?

**Review by amyeroberts — COMMENTED** [review]
Thanks for fixing! 

Do you know why this breaking change wasn't caught on the optimum side? 
Can you explain a bit more how these changes fix the problem? The changes look OK but it's not clear how they link to the issue description.

File: tests/quantization/gptq/test_gptq.py
Comment (SunMarc) [reply tests/quantization/gptq/test_gptq.py:250]: With this [PR](https://github.com/huggingface/optimum/pull/1574/files#), we don't save all the arguments anymore (only those in `self.serialization_keys`) by modifying `to_dict()` . The issue with that is that we are updating the quantization_config based on the one from optimum: `config.quantization_config = GPTQConfig.from_dict_optimum(quantizer.to_dict())`. 
This line was needed since some args could change like `use_exllama` in optimum. 

I was thinking on doing a PR to remove this line, and maybe not save args related to inference anymore (use_exllama,...) or revert the PR on optimum. What are your thoughts ?

File: tests/quantization/gptq/test_gptq.py
Comment (SunMarc) [reply tests/quantization/gptq/test_gptq.py:245]: see above

File: tests/quantization/gptq/test_gptq.py
Comment (amyeroberts) [reply tests/quantization/gptq/test_gptq.py:250]: > This line was needed since some args could change like use_exllama in optimum.

Sorry, I don't completely follow. Does this mean that `use_exllama` will no longer change and the test check is no longer required. 

> I was thinking on doing a PR to remove this line, and maybe not save args related to inference anymore (use_exllama,...) 

It depends. This can be considered a breaking change, as users might now expect these values in their configs. The most important thing is for old configs to still be loadable and produce the same result.

Comment (SunMarc) [reply tests/quantization/gptq/test_gptq.py:250]: > Sorry, I don't completely follow. Does this mean that use_exllama will no longer change and the test check is no longer required?

Basically, I mean that the user can set `use_exllama=True` in transformers and this value can change in optimum (`use_exllama=False`). However, since we don't serialize it anymore in optimum GPTQ config, `use_exllama` will be set to the default value through: `config.quantization_config = GPTQConfig.from_dict_optimum(quantizer.to_dict()).`

> It depends. This can be considered a breaking change, as users might now expect these values in their configs. The most important thing is for old configs to still be loadable and produce the same result.

Yes, the old configs will still work. However, for new users, they will have to pass these args each time. 

I will probably work on the second option then since from the start, we should not have to let the user select the kernel since we can switch from one to another.

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing the tests and explaining! 

I'm approving as this is effectively just fixing the tests: the changes to GPTQ config have already been made and we haven't experience lots of users complaining. However, making changes like this can be breaking and cause a lot of issues downstream. Because of the coupling between optimum and the respective configs in transformers I'd expect both for changes in API to have backwards compatibility as part of their consideration, and for there to be sufficient tests run to make sure changes in optimum that affect transformers are caught before they're merged in.

**Comment by amyeroberts** [comment]
@SunMarc Do you have permissions to merge? If not, I can merge this in if it's good to go

**Pull request merged**
